causallearn/utils/cit.py

```diff
@@ -31,10 +31,13 @@ def __init__(self, data, method='fisherz', **kwargs):
 
         if method == 'kci':
             # parse kwargs contained in the KCI method
-            kci_kwargs = {k: v for k, v in kwargs.items() if k in
-                          ['kernelX', 'kernelY', 'null_ss', 'approx', 'est_width', 'polyd', 'kwidthx', 'kwidthy']}
-            self.kci_ui = KCI_UInd(**kci_kwargs)
-            self.kci_ci = KCI_CInd(**kci_kwargs)
+            kci_ui_kwargs = {k: v for k, v in kwargs.items() if k in
+                             ['kernelX', 'kernelY', 'null_ss', 'approx', 'est_width', 'polyd', 'kwidthx', 'kwidthy']}
+            kci_ci_kwargs = {k: v for k, v in kwargs.items() if k in
+                             ['kernelX', 'kernelY', 'kernelZ', 'null_ss', 'approx', 'use_gp', 'est_width', 'polyd',
+                              'kwidthx', 'kwidthy', 'kwidthz']}
+            self.kci_ui = KCI_UInd(**kci_ui_kwargs)
+            self.kci_ci = KCI_CInd(**kci_ci_kwargs)
```

Comment:
hmmm, seems a bug to me previously.

Are the old runnable when calling KCI_CInd?

Reply:
My fault! The old code is runnable when calling KCI_CInd, but not KCI_UInd, since there are arguments e.g., `kernelZ`, `use_gp` specifically for KCI_CInd and there is no **kwargs in KCI_UInd.

To prevent from this again, I'll add a unit test using KCI in `TestPC.py` soon.

Reply:
awesome, thanks!!